### PR TITLE
[NFC][SYCL] Update tests which were not properly working for Windows

### PR DESCRIPTION
**Pull request by mdtoguchi**
A few of the Driver sycl* tests were not configured properly to work
in a windows environment and were disabled.  Make the needed adjustments
to the tests and break out any specific testing that needs Linux and
Windows specificity.

### CODE REVIEW

**Comment by mdtoguchi** [comment]
This work is to address https://github.com/intel/llvm/issues/888

File: clang/test/Driver/sycl-intelfpga-aoco.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// CHK-FPGA-AOCO-PHASES: 17: llvm-spirv, {16}, spirv, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 18: input, "[[INPUTA]]", archive
+// CHK-FPGA-AOCO-PHASES: 19: clang-offload-unbundler, {18}, fpga_dependencies_list
+// CHK-FPGA-AOCO-PHASES: 20: input, "[[INPUTA]]", fpga_aoco
+// CHK-FPGA-AOCO-PHASES: 21: clang-offload-unbundler, {20}, fpga_aoco
+// CHK-FPGA-AOCO-PHASES: 22: backend-compiler, {17, 19, 21}, fpga_aocx, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 23: clang-offload-wrapper, {22}, object, (device-sycl)
+// CHK-FPGA-AOCO-PHASES: 24: offload, "host-sycl (x86_64-unknown-linux-gnu)" {10}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {23}, image
+
+/// FPGA AOCO Windows phases check
```
Comment (AGindinson) [inline clang/test/Driver/sycl-intelfpga-aoco.cpp:44]: Would these really be needed in a Linux-specific file?

File: clang/test/Driver/sycl-offload-win.c [hunk 2]
```diff
@@ -86,20 +85,18 @@
 
 /// ###########################################################################
 
-// RUN: touch %t.lib
-// RUN: %clang --target=x86_64-pc-windows-msvc -fsycl -foffload-static-lib=%t.lib %s -### 2>&1 \
-// RUN:   | FileCheck -DLIB=%t.lib %s -check-prefix=FOFFLOAD_STATIC_LIB_SRC2
-// RUN: %clang_cl --target=x86_64-pc-windows-msvc -fsycl -foffload-static-lib=%t.lib %s -### 2>&1 \
-// RUN:   | FileCheck -DLIB=%t.lib %s -check-prefix=FOFFLOAD_STATIC_LIB_SRC2
-// FOFFLOAD_STATIC_LIB_SRC2: clang-offload-bundler{{(.exe)?}}{{.+}} "-type=aoo" "-targets=sycl-spir64-{{.+}}-sycldevice" "-inputs=[[LIB]]"{{.+}} "-unbundle"
+// RUN: touch %t-orig.lib
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload-win.c:88]: How exactly does such renaming help with Windows path issues?

File: clang/test/Driver/sycl-intelfpga-aoco.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-intelfpga-aoco.cpp:44]: It tests the use of `clang-cl` which is allowed on the Linux side of testing.  So while it's not really necessary to test on Linux, it does provide some testing for the Windows path for those who do not do testing on Windows when doing local validation of changes.

File: clang/test/Driver/sycl-offload-win.c (hunk 2 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload-win.c:88]: The problem we are dealing with is using `-DVAR=%t`.  As `%t` contains full directory information, this isn't translating well when being passed in.  The directory separators are not consistent causing miscompares, so I'm using additional strings in the file names to track within the output.

**Review by AGindinson — APPROVED** [review]
LGTM! A minor suggestion would be to add descriptive headers to `sycl-intelfpga-aoco*.cpp` files.

**Pull request merged**